AppCenterAnalytics/AppCenterAnalyticsTests/MSPropertyConfiguratorTests.m

```diff
@@ -0,0 +1,83 @@
+#import <XCTest/XCTest.h>
+
+#import "MSChannelGroupProtocol.h"
+#import "MSCommonSchemaLog.h"
+#import "MSPropertyConfiguratorPrivate.h"
+#import "MSTestFrameworks.h"
+#import "MSAnalyticsTransmissionTargetInternal.h"
+
+@interface MSPropertyConfiguratorTests : XCTestCase
+
+@property(nonatomic) MSPropertyConfigurator *sut;
+@property(nonatomic) MSAnalyticsTransmissionTarget *transmissionTarget;
+@property(nonatomic) id configuratorClassMock;
+
+@end
+
+@implementation MSPropertyConfiguratorTests
+
+- (void)setUp {
+  [super setUp];
+  
+  self.sut = [MSPropertyConfigurator new];
+
+  // Mock the init so that self.sut can be injected into the target.
+  self.configuratorClassMock = OCMClassMock([MSPropertyConfigurator class]);
+  OCMStub([self.configuratorClassMock alloc]).andReturn(self.configuratorClassMock);
+  OCMStub([self.configuratorClassMock initWithTransmissionTarget:OCMOCK_ANY]).andReturn(self.sut);
+  id channelGroupMock = OCMProtocolMock(@protocol(MSChannelGroupProtocol));
+  self.transmissionTarget = OCMPartialMock([[MSAnalyticsTransmissionTarget alloc] initWithTransmissionTargetToken:@"123" parentTarget:nil channelGroup:channelGroupMock]);
+  OCMStub([self.transmissionTarget isEnabled]).andReturn(YES);
+  self.sut.transmissionTarget = self.transmissionTarget;
+}
+
+- (void)tearDown {
+  [super tearDown];
+  self.sut = nil;
+  [self.configuratorClassMock stopMocking];
+}
+
+- (void)testInitializationWorks {
+  
+  // Then
+  XCTAssertNotNil(self.sut);
+  XCTAssertFalse(self.sut.shouldCollectDeviceId);
+}
+
+- (void)testCollectDeviceIdWorks {
+  
+  // When
+  [self.sut collectDeviceId];
+  
+  // Then
+  XCTAssertTrue(self.sut.shouldCollectDeviceId);
+}
+
+- (void)testCollectsDeviceIdWhenShouldCollectDeviceIdIsTrue {
+#if !TARGET_OS_OSX
```

Comment:
More complex and not technically in scope, so we decided to follow the same pattern as for other macOS 1DS features and provide a rough implementation but not worry about testing (we didn't add to macOS test app either)

Reply:
I disagree. No matter what intentions or behind stories are, we shouldn't lose any code coverage because of this one. Writing unit tests based on current implementation should be enough.